### DISCUSSION TITLE
Is anyone here really familiar with the docker-compose setup for this?

### DISCUSSION THREAD

**Discussion by Jaffythethird** [itzg/docker-minecraft-server#2142] [General]
I'll admit, I'm new to Docker. That being said, I _think_ I've got enough figured out to be able to start and stop a server with a custom port. However, the server seems to be getting slowed down any time that more than one person is on, to the point that you can't pick anything up and it disconnects you. I'd like to think it's high resource usage, but I cannot for the life of me figure out how to troubleshoot it from outside the container. 
I've tried sending rcon-cli commands through docker-compose, but I would get no response and echoing $? returned a 1 every time. I'm 100% convinced that this is a fault of my not understanding docker-compose, and that this would all work correctly if I was doing it right. However, I think I've gotten as far as I can with this via documentation without calling for a little help from someone. Anyone volunteer to help? I can try and fetch any info that you need.

**Comment by anaxios** [comment]
While the server container is running you can `docker attach <container name>` to interact with the server console.

**Reply by Jaffythethird** [reply]
> While the server container is running you can `docker attach <container name>` to interact with the server console.

Tried it and it isn't interactive. If it's attached, I can see logs it puts out, but I can't do anything with it. Thought it was due to not having the -it flags upon running, but that didn't work with compose.

**Reply by anaxios** [reply]
Do you have this environment variable set? `-e EXEC_DIRECTLY=true`

**Reply by Jaffythethird** [reply]
> Do you have this environment variable set? `-e EXEC_DIRECTLY=true`

I am certain that I do not. For the purposes of docker-compose, should this be set in the YAML or does it have to be done in the up command?

**Reply by Jaffythethird** [reply]
I managed to run the op command now using `docker exec mc rcon-cli op USER`. I had been using docker-compose as I had had problems running docker standalone commands earlier and only docker-compose worked, and I had thought the documentation did not cover that circumstance. I believe this helps me get the ball rolling. I did also set that variable to true, which may have helped as well. Either way, thank you. I would not have figured this out without your help

**Reply by anaxios** [reply]
I recommend finding a docker-compose tutorial. That will help. You can set environment variables in the compose yaml.

**Reply by itzg** [reply]
And there's plenty of examples here, if you hadn't seen them yet:

https://github.com/itzg/docker-minecraft-server/tree/master/examples

The "simple" one shows how TTY and stdin can be enabled:

https://github.com/itzg/docker-minecraft-server/blob/master/examples/docker-compose-simple.yml#L6-L7
